core/shared/src/main/scala/zio/stm/TRef.scala

```diff
@@ -35,10 +36,20 @@ import java.util.concurrent.atomic.AtomicReference
  * do not support concurrent access.
  */
 final class TRef[A] private (
-  @volatile private[stm] var versioned: Versioned[A],
-  private[stm] val todo: AtomicReference[Map[TxnId, Todo]]
+  private[stm] val versioned: AtomicReference[A],
+  @volatile private[stm] var todo: Map[TxnId, Todo] = Map.empty
 ) extends Serializable { self =>
 
+  @deprecated("kept for binary compatibility only", "2.1.8")
+  def this(versioned: Versioned[A], todo: AtomicReference[Map[TxnId, Todo]]) =
+    this(new AtomicReference(versioned.value), todo.get())
+
+  private[stm] val lock: ZSTMLockSupport.Lock = ZSTMLockSupport.Lock(true)
+
+  private[this] val hc = super.hashCode()
```

Comment:
can drop `[this]`?

Reply:
The Scala 2 `private` and `private[this]` generate different bytecode. When `[this]` is not used, the compiler will create accessor methods for the val so that other instances of the class can access it. e.g., the code below:

```scala
class Hello {
  private val foo = "hello"
  private[this] val bar = "world"
}
```
<details>
  <summary>Produces the following bytecode (Click to expand)</summary>

```
// class version 52.0 (52)
// access flags 0x21
public class Hello {

  // compiled from: Main.scala

  @Lscala/reflect/ScalaSignature;(bytes="\u0006\u0005\u00012A!\u0002\u0004\u0001\u0013!)\u0001\u0003\u0001C\u0001#!9A\u0003\u0001b\u0001\n\u0013)\u0002B\u0002\u0010\u0001A\u0003%a\u0003\u0003\u0004 \u0001\u0001\u0006IA\u0006\u0002\u0006\u0011\u0016dGn\u001c\u0006\u0002\u000f\u00059A(Z7qift4\u0001A\n\u0003\u0001)\u0001\"a\u0003\u0008\u000e\u00031Q\u0011!D\u0001\u0006g\u000e\u000cG.Y\u0005\u0003\u001f1\u0011a!\u00118z%\u00164\u0017A\u0002\u001fj]&$h\u0008F\u0001\u0013!\u0009\u0019\u0002!D\u0001\u0007\u0003\r1wn\\\u000b\u0002-A\u0011q\u0003H\u0007\u00021)\u0011\u0011DG\u0001\u0005Y\u0006twMC\u0001\u001c\u0003\u0011Q\u0017M^1\n\u0005uA\"AB*ue&tw-\u0001\u0003g_>\u0004\u0013a\u00012be\u0002")

  ATTRIBUTE ScalaSig : unknown

  ATTRIBUTE ScalaInlineInfo : unknown

  // access flags 0x12
  private final Ljava/lang/String; foo

  // access flags 0x12
  private final Ljava/lang/String; bar

  // access flags 0x2
  private foo()Ljava/lang/String;
   L0
    LINENUMBER 2 L0
    ALOAD 0
    GETFIELD Hello.foo : Ljava/lang/String;
    ARETURN
   L1
    LOCALVARIABLE this LHello; L0 L1 0
    MAXSTACK = 1
    MAXLOCALS = 1

  // access flags 0x1
  public <init>()V
   L0
    LINENUMBER 1 L0
    ALOAD 0
    INVOKESPECIAL java/lang/Object.<init> ()V
   L1
    LINENUMBER 2 L1
    ALOAD 0
    LDC "hello"
    PUTFIELD Hello.foo : Ljava/lang/String;
   L2
    LINENUMBER 3 L2
    ALOAD 0
    LDC "world"
    PUTFIELD Hello.bar : Ljava/lang/String;
   L3
    LINENUMBER 1 L3
    RETURN
   L4
    LOCALVARIABLE this LHello; L0 L4 0
    MAXSTACK = 2
    MAXLOCALS = 1
}
```
</details>

(see how there is a `private foo()` method but not `private bar()`)

Reply:
@kyri-petrou is private[this] not deprecated in scala 3? 🤔

Reply:
> @kyri-petrou is private[this] not deprecated in scala 3? 🤔

@987Nabil It depends on what `-source` scalac flag you use. It raises a compiler warning if you're using `-source:future` but doesn't if you leave it unspecified